lib/mutant/mutator/node/regexp.rb

```diff
@@ -0,0 +1,104 @@
+module Mutant
+  class Mutator
+    class Node
+
+      # Generic mutator
+      module RegularExpression
+        class Parser
+          def self.parse(source)
+            Node.new(Regexp::Parser.parse(source, 'ruby/2.3'))
+          end
+
+          class Node < DelegateClass(Regexp::Expression::Base)
```

Comment:
^ Also to clarify on the above comment: I absolutely agree that it is worth it now that I have had to heavily use the mutable `Regexp::Expression` mutable API. I just found it to be way more work than I anticipated